backend/src/recommitment/recommitment.service.ts

```diff
@@ -0,0 +1,425 @@
+      await this.recommitmentRepository.update(
+        {
+          personnelId: id,
+          recommitmentCycleId: recommitmentUpdate[key].year,
+          program: recommitmentUpdate[key].program,
+        },
+        {
+          ...recommitment,
+          status: recommitmentUpdate[key].status,
+        },
+      );
+      this.triggerEmailNotification(id, recommitmentUpdate[key]);
+    }
+
+    return await this.personnelService.findOne(id);
+  }
+  /**
+   * Send email after a trigger (ie: member accepted, supervisor denied, etc)
+   * @param id
+   * @param recommitmentUpdate
+   * @param key
+   */
+  async triggerEmailNotification(
+    id: string,
+    recommitmentUpdate: UpdatePersonnelRecommitmentDTO,
+  ): Promise<void> {
+    const personnel = await this.recommitmentRepository.findOneOrFail({
+      where: { personnelId: id },
+      relations: ['personnel'],
+    });
+    const recommitmentCycle = await this.checkRecommitmentPeriod();
+    switch (recommitmentUpdate.status) {
+      case RecommitmentStatus.MEMBER_COMMITTED:
+        return await this.mailService.sendMail(
+          this.mailService.generateTemplate(
+            EmailTemplates.SUPERVISOR_REQUEST,
+            TemplateType.SUPERVISOR,
+            [personnel.toResponseObject()],
+            recommitmentCycle.endDate,
+            recommitmentUpdate.program,
+          ),
+        );
+
+      case RecommitmentStatus.MEMBER_DENIED:
+        return await this.mailService.sendMail(
+          this.mailService.generateTemplate(
+            EmailTemplates.MEMBER_DECLINED,
+            TemplateType.MEMBER,
+            [personnel.toResponseObject()],
+            recommitmentCycle.endDate,
+            recommitmentUpdate.program,
+          ),
+        );
+      case RecommitmentStatus.SUPERVISOR_APPROVED:
+        return await this.mailService.sendMail(
+          this.mailService.generateTemplate(
+            EmailTemplates.MEMBER_APPROVED,
+            TemplateType.MEMBER,
+            [personnel.toResponseObject()],
+            recommitmentCycle.endDate,
+            recommitmentUpdate.program,
+          ),
+        );
+
+      case RecommitmentStatus.SUPERVISOR_DENIED:
+        return await this.mailService.sendMail(
+          this.mailService.generateTemplate(
+            EmailTemplates.MEMBER_DENIED,
+            TemplateType.MEMBER,
+            [personnel.toResponseObject()],
+            recommitmentCycle.endDate,
+            recommitmentUpdate.program,
+          ),
+        );
+    }
+  }
+
+  /**
+   * Checks if the current date is within the recommitment period.
+   * @returns {Promise<RecommitmentCycleRO>} - The recommitment cycle information.
+   */
+  async checkRecommitmentPeriod(): Promise<RecommitmentCycleRO> {
+    const qb = this.recommitmentCycleRepository.createQueryBuilder();
+    qb.where('start_date <= :date', { date: new Date() });
+    qb.andWhere('end_date >= :date', { date: new Date() });
```

Comment:
We may run into timezone issues here..

Reply:
I think we should be okay since the start date/end date for the table are generated on the BE but you are right to flag this. I'll bring this up with steffi during the testing discussion.